libsolidity/ast/Types.cpp

```diff
@@ -1529,7 +1529,7 @@ void ArrayType::clearCache() const
 BoolResult ArrayType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 {
 	if (_convertTo.category() != category())
-		return false;
+		return isByteArray() && _convertTo.category() == Type::Category::FixedBytes;
```

Comment:
Do we really want to allow implicit conversion? Isn't that rather dangerous?

Reply:
Hm, I was thinking that was how we wanted to do it. But, I see it can be dangerous.
So, I can change to explicit conversion instead then?

Reply:
Also I don't think we should support `string -> bytesNN` without going through `bytes`.

Reply:
Allowed only explicit conversion and disallowed `string -> bytesNN` without going through `bytes`